src/main/java/com/ksabbak/javaserver/server/request/RequestHandler.java

```diff
@@ -0,0 +1,59 @@
+package com.ksabbak.javaserver.server.request;
+
+import com.ksabbak.javaserver.router.Router;
+import com.ksabbak.javaserver.server.Response;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.Socket;
+
+public class RequestHandler implements Runnable{
```

Comment:
This class is a very good step in the right direction. I like this class.

Conceptually, this feels like a single connection (especially given that it takes a `Socket`). I would consider renaming it to something that includes "Connection" in the same, e.g. `HttpConnection`, `IncomingConnection`, `Connection`, etc.

One further step that might help improve this class even more is to create something that wraps `Socket` more directly, and pass that wrapper into here instead of passing `Socket` in. This idea would be easier to explain in person or over a call. I'm also not sure you have time for it.

Reply:
Thanks for the name suggestion!